projects/plugins/jetpack/_inc/lib/class-jetpack-podcast-helper.php

```diff
@@ -236,6 +247,25 @@ public static function set_podcast_locator( &$feed ) {
 		$feed->set_locator_class( 'Jetpack_Podcast_Feed_Locator' );
 	}
 
+	/**
+	 * Action handler to reset the SimplePie cache for the podcast feed.
+	 *
+	 * Note this only resets the cache for the specified url. If the feed locator finds the podcast feed
+	 * within the markup of the that url, that feed itself may still be cached.
+	 *
+	 * @param SimplePie $feed The SimplePie object, passed by reference.
+	 * @return void
+	 */
+	public static function reset_simplepie_cache( &$feed ) {
+		// Retrieve the cache object for a feed url. Based on:
+		// https://github.com/WordPress/WordPress/blob/fd1c2cb4011845ceb7244a062b09b2506082b1c9/wp-includes/class-simplepie.php#L1412.
+		$cache = $feed->registry->call( 'Cache', 'get_handler', array( $feed->cache_location, call_user_func( $feed->cache_name_function, $feed->feed_url ), 'spc' ) );
```

Comment:
I was curious, so I tested this with a feed that redirects and it seems that the URL used in the cache is the given URL, not the one it redirects to. 

Similarly, if the URL isn't a feed, but a feed is located on the page, it looks as though the cache key is based in on the original URL and not the eventual feed URL.

Reply:
@pablinos To clarify, does that mean this will bust the feed cache, as long as the user retries with the same url (even if the url is a redirect)?

Reply:
Yes, that's right, it appears to store the content against the initial URL. My concern was that it would store it against the redirected (or otherwise found) URL, so this call to `unlink` would have no effect.